### PR TITLE
BigQueryIO: move write validation to validate() from apply()

### PR DESCRIPTION
**Pull request by dhalperi**
If runners override the transform, validation will still happen.

### CODE REVIEW

**Comment by dhalperi** [comment]
R: @peihe

File: sdks/java/core/src/main/java/org/apache/beam/sdk/io/BigQueryIO.java [hunk 1]
```diff
@@ -963,8 +960,8 @@ public PDone apply(PCollection<TableRow> input) {
         // Note that a presence check can fail if the table or dataset are created by earlier stages
         // of the pipeline. For these cases the withoutValidation method can be used to disable
         // the check.
-        // Unfortunately we can't validate anything early in case tableRefFunction is specified.
-        if (table != null && validate) {
+        // Unfortunately we can't validate anything early if tableRefFunction is specified.
+        if (jsonTableRef != null && validate) {
```
Comment (peihe) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/io/BigQueryIO.java]: why change it from table?
table is used in the if block

File: sdks/java/core/src/main/java/org/apache/beam/sdk/io/BigQueryIO.java [hunk 2]
```diff
@@ -986,24 +982,38 @@ public PDone apply(PCollection<TableRow> input) {
             throw new IllegalArgumentException("WriteDisposition.WRITE_TRUNCATE is not "
                 + "supported for unbounded PCollections or when using tablespec functions.");
           }
-
-          return input.apply(new StreamWithDeDup(table, tableRefFunction, getSchema()));
-        }
-
+        } else {
+          // We will use a BigQuery load job -- validate the temp location.
         String tempLocation = options.getTempLocation();
```
Comment (peihe) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/io/BigQueryIO.java]: indents, and below

**Comment by dhalperi** [comment]
Thanks, PTAL @peihe

**Comment by peihe** [comment]
LGTM

**Pull request closed without merging**